### DISCUSSION TITLE
File-structure as source

### DISCUSSION THREAD

**Discussion by jgollenz** [nvim-neo-tree/neo-tree.nvim#229] [Ideas]
What do you think of showing the structure of the current buffer as a tree for overview purposes? Like the sidebar on a PDF reader. I'm working on a large Latex file and I think it would be great to quickly get an overview of where I am and where I want to jump to next. One could also think of it as a tree of all collapse-able sections in a source file. I assume this can be achieved with Treesitter.

**Comment by cseickel** [comment]
I think it would be great to have. I don't personally work with Latex (yet) so I can't take this one on, but I'd be happy to merge a PR for this.

I'm not sure if it belongs in the same source or not, but this would also be nice to have for markdown and vim help files.

**Reply by jgollenz** [reply]
I think it would work well for any file that has a tree like structure. When you think about it, Treesitter does this for any source file. "nvim-treesitter/playground" does something like what I'm thinking of, but it shows the whole tree. I think there is no need to go down deeper than functions, but just think how useful something like that would be in a file with lots of functions
